rust/src/detect/iprep.rs

```diff
@@ -75,36 +76,46 @@ extern "C" {
     pub fn SRepCatGetByShortname(name: *const c_char) -> u8;
 }
 
-pub fn detect_parse_iprep(i: &str) -> IResult<&str, DetectIPRepData> {
-    let (i, _) = opt(is_a(" "))(i)?;
-    let (i, cmd) = map_res(alpha0, DetectIPRepDataCmd::from_str)(i)?;
-    let (i, _) = opt(is_a(" "))(i)?;
-    let (i, _) = char(',')(i)?;
-    let (i, _) = opt(is_a(" "))(i)?;
+pub fn detect_parse_iprep(i: &str) -> IResult<&str, DetectIPRepData, RuleParseError<&str>> {
+    // Inner utility function for easy error creation.
+    fn make_error(reason: String) -> nom7::Err<RuleParseError<&'static str>> {
+        Err::Error(RuleParseError::InvalidIPRep(reason))
+    }
+    let (_, values) = nom7::multi::separated_list1(
+        tag(","),
+        preceded(multispace0, nom7::bytes::complete::is_not(",")),
```

Comment:
Why this change ?

This is normally used if you do not care about the subparts of a keyword

Reply:
Guess I'm not really sure what other ways there are to do it. I just want to split by `,` or end of string.

Reply:
btw took this directly from the bytemath parsing code, even though that has a slightly different layout

Reply:
> btw took this directly from the bytemath parsing code

bytemath is right using this, because bytemath is a list of sub keywords like `endian` or `relative`, whose order is not relevant. bytemath parses kind of a list of key+values.

Here, you want to parse something different, you parse a list of specific things...

> I just want to split by , or end of string.

I think it is easier to have some linear code like currently, but yes, if you really want to first split into words/fields, this is good.

But then, you should not duplicate code, cf below...